### PR TITLE
feat: use any person property for simple filters

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

I was replying to a support ticket suggesting they set "phone number" as a person display name property. It felt wrong that simple filters weren't more customizable.

## Changes

- Use the existing `PropertyKeyInfo` component to render property names
- Allow customization of person properties in the simple filters (stored in persisted logic per user)

## How did you test this code?

https://github.com/PostHog/posthog/assets/6685876/2ca0666e-b347-4da3-8bd1-14b314dcc15b

### CODE REVIEW

File: frontend/src/scenes/session-recordings/player/playerSettingsLogic.ts [hunk 1]
```diff
@@ -204,6 +205,15 @@ export const playerSettingsLogic = kea<playerSettingsLogicType>([
...
         ],
+        quickFilterProperties: [
+            ['$geoip_country_name'] as string[],
+            {
+                persist: true,
+            },
+            {
+                setQuickFilterProperties: (_, { properties }) => properties,
+            },
+        ],
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/playerSettingsLogic.ts:216]: this is really cool!

i can see the next question being "i set up my quick filters then they don't show on my desktop" :p

but that's the overlap with Ben's saved filters suggestion

File: frontend/src/scenes/session-recordings/filters/SimpleSessionRecordingsFilters.tsx [hunk 2]
```diff
@@ -122,29 +123,97 @@ export const SimpleSessionRecordingsFilters = ({
...
 }
+
+const Configuration = ({
+    properties,
+    onChange,
+}: {
+    properties: string[]
+    onChange: (properties: string[]) => void
+}): JSX.Element => {
+    // const buttonRef = useRef<HTMLButtonElement>(null)
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/filters/SimpleSessionRecordingsFilters.tsx:176]: commenty be gone!
```suggestion
```

File: frontend/src/scenes/session-recordings/filters/SimpleSessionRecordingsFilters.tsx [hunk 3]
```diff
@@ -122,29 +123,97 @@ export const SimpleSessionRecordingsFilters = ({
...
+                            setShowPropertySelector(false)
+                        }}
+                        taxonomicGroupTypes={[TaxonomicFilterGroupType.PersonProperties]}
+                        excludedProperties={{ [TaxonomicFilterGroupType.PersonProperties]: properties }}
+                    />
+                }
+            >
+                <LemonButton onClick={() => setShowPropertySelector(!showPropertySelector)} fullWidth>
+                    Add person properties
+                </LemonButton>
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/filters/SimpleSessionRecordingsFilters.tsx:215]: could we add event properties as an option?

File: frontend/src/scenes/session-recordings/filters/SimpleSessionRecordingsFilters.tsx [hunk 4]
```diff
@@ -122,29 +123,97 @@ export const SimpleSessionRecordingsFilters = ({
...
-                    </LemonButton>
-                </LemonMenu>
+                    <LemonMenu
+                        items={[
+                            defaultItems.length > 0 && { items: defaultItems },
+                            personPropertyItems.length > 0 && {
+                                title: 'Person properties',
+                                items: personPropertyItems,
+                            },
+                        ]}
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/filters/SimpleSessionRecordingsFilters.tsx:147]: then we could label the items in the menu with the person or event icon so people can see what they have a little?

**Review by pauldambra — APPROVED** [review]
I really like this

A few suggestions, but I'd be tempted to say to get this in and follow-up

**Review by marandaneto — APPROVED** [review]
Noice!

**Pull request merged**